tests/Unit/Emogrifier/HtmlProcessor/CssInlinerTest.php

```diff
@@ -1281,6 +1281,87 @@ public function emogrifyMatchesRuleAfterUnneededCssThing($css)
         static::assertContains('<body style="color: green;">', $result);
     }
 
+    /**
+     * Processing of @media rules may involve removal of some unnecessary whitespace from the CSS placed in the <style>
+     * element added to the docuemnt, due to the way that certain parts are `trim`med.  Notably, whitespace either side
+     * of "{" and "}" may be removed.
+     *
+     * This method helps takes care of that, by converting a search needle for an exact match into a regular expression
+     * that allows for such whitespace removal, so that the tests themselves do not need to be written less humanly
+     * readable and can use inputs containing extra whitespace.
+     *
+     * @param string $needle Needle that would be used with `assertContains` or `assertNotContains`
+     *
+     * @return string Needle to use with `assertRegExp` or `assertNotRegExp` instead
+     */
+    private static function getNeedleRegExpAllowingRemovalOfUnnecessaryCssWhitespace($needle)
+    {
+        $needleMatcher = preg_replace_callback(
+            '/\\s*+([{}])\\s*+|(?:(?!\\s*+[{}]).)++/',
+            function ($matches) {
```

Comment:
`$matches` can have the `array` type hinting.

Reply:
Added.  I note I missed a type hint in the previous PR which you added before committing.  I've made a mental note to double-check these in future.